### PR TITLE
dep(dm): fix panic when resume task which is in dump stage

### LINKED ISSUE #5970: DM: dm panic when resume task which is in dump stage
### What did you do?

resume task which is in dump stage

### What did you expect to see?

normal resume

### What did you see instead?

```
panic: duplicate metrics collector registration attempted

goroutine 3375 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc001532fa0?, {0xc00f74f290, 0x1, 0x68f42fc?})
        /home/cuixiaoyuan/go/pkg/mod/github.com/prometheus/client_golang@v1.12.2/prometheus/registry.go:403 +0xa7
github.com/pingcap/tidb/dumpling/export.(*metrics).registerTo(0xc004e9ac40, {0x7ec3440, 0xc0005ac910})
        /home/cuixiaoyuan/go/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20220617092035-96526518b4e8/dumpling/export/metrics.go:96 +0xce
github.com/pingcap/tidb/dumpling/export.NewDumper({0x7ec9148, 0xc004e9abc0}, 0xc000f49180)
        /home/cuixiaoyuan/go/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20220617092035-96526518b4e8/dumpling/export/dump.go:85 +0x43c
github.com/pingcap/tiflow/dm/dumpling.(*Dumpling).Process(0xc00409cff0, {0x7ec9148, 0xc0041d3f00}, 0xc004190a80?)
        /home/cuixiaoyuan/workspace/enhance-dmctl-task/dm/dumpling/dumpling.go:163 +0xa5b
github.com/pingcap/tiflow/dm/dumpling.(*Dumpling).Resume(0xc00409cff0, {0x7ec9148, 0xc0041d3f00}, 0x179351a?)
        /home/cuixiaoyuan/workspace/enhance-dmctl-task/dm/dumpling/dumpling.go:244 +0xde
created by github.com/pingcap/tiflow/dm/dm/worker.(*SubTask).Resume
        /home/cuixiaoyuan/workspace/enhance-dmctl-task/dm/dm/worker/subtask.go:651 +0x6bc
```

### Versions of the cluster

master

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Issue Number: close #5970

### What is changed and how it works?
as the analysis of causes in [#5970](https://github.com/pingcap/tiflow/issues/5970), dm should update tidb refrence. 
because the reason is clear, and the timing of termination is not easy to control in integrate test, we just add some warn log and update tidb refrence.

#### Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/dumpling/dumpling.go [hunk 1]
```diff
@@ -171,6 +171,8 @@ func (m *Dumpling) Process(ctx context.Context, pr chan pb.ProcessResult) {
 			m.logger.Info("", zap.String("failpoint", "SleepBeforeDumplingClose"))
 		})
 		dumpling.Close()
+	} else {
+		log.L().Warn("error occurred during NewDumper", zap.Error(err))
```
Comment (lance6716) [inline dm/dumpling/dumpling.go:175]: should use m.logger

**Review by lance6716 — APPROVED** [review]

File: dm/dumpling/dumpling.go (hunk 1 above)
Comment (WizardXiao) [reply dm/dumpling/dumpling.go]: ok

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**